src/Phoenix/PhoenixClient.ts

```diff
@@ -177,4 +180,61 @@ export class PhoenixClient {
       [HttpHeaders.contentType]: "application/json",
     };
   }
+
+  public showForbiddenFailurePopUp(connectionInfoData: IValidationError) {
+    const errInfo = connectionInfoData;
+    switch (errInfo?.type) {
+      case "MaxAllocationTimeExceeded": {
+        const maxAllocationTimeExceeded = errInfo as IMaxAllocationTimeExceeded;
+        const allocateAfterTimestamp = new Date(maxAllocationTimeExceeded?.earliestAllocationTimestamp);
+        allocateAfterTimestamp.setDate(allocateAfterTimestamp.getDate() + 1);
+        useDialog
+          .getState()
+          .showOkModalDialog(
+            "Connection Failed",
+            `${errInfo.message}` +
+              "Max allocation time for a day to a user is " +
+              `${maxAllocationTimeExceeded.maxAllocationTimePerDayPerUserInMinutes}` +
+              ". Please try again after " +
+              `${allocateAfterTimestamp.toLocaleString()}`
+          );
+        break;
+      }
+      case "MaxDbAccountsPerUserExceeded": {
+        const maxDbAccountsPerUserExceeded = errInfo as IMaxDbAccountsPerUserExceeded;
+        useDialog
+          .getState()
+          .showOkModalDialog(
+            "Connection Failed",
+            `${errInfo.message}` +
+              " Max simultaneous connections allowed per user is " +
+              `${maxDbAccountsPerUserExceeded.maxSimultaneousConnectionsPerUser}` +
+              "."
+          );
+        break;
+      }
+      case "MaxUsersPerDbAccountExceeded": {
+        const maxUsersPerDbAccountExceeded = errInfo as IMaxUsersPerDbAccountExceeded;
+        useDialog
+          .getState()
+          .showOkModalDialog(
+            "Connection Failed",
+            `${errInfo.message}` +
+              "Max simultaneous users allowed per DbAccount is " +
+              `${maxUsersPerDbAccountExceeded.maxSimultaneousUsersPerDbAccount}` +
+              "."
+          );
+        break;
+      }
+      case "AllocationValidationResult":
+      case "RegionNotServicable":
+      case "SubscriptionNotAllowed": {
+        useDialog.getState().showOkModalDialog("Connection Failed", `${errInfo.message}`);
```

Comment:
why not have a string "errorMessage" and set it in the switch case block, and finally do useDialog.getState().showOkModalDialog()?